### ISSUE TITLE
Joomla\CMS\Installer\InstallerScript::getInstances Unknown column 'id' in 'field list'

### ISSUE THREAD

**Issue by svanschu** [joomla/joomla-cms#42191]
In this method it tries to load the field 'id' from the extensions table. But there isn't any field called id in this table.

The id field for the extenions table is called extension_id
```
        // Select the item(s) and retrieve the id
        $query->select($db->quoteName('id'));

        if ($isModule) {
            $query->from($db->quoteName('#__modules'))
                ->where($db->quoteName('module') . ' = :extension');
        } else {
            $query->from($db->quoteName('#__extensions'))
                ->where($db->quoteName('element') . ' = :extension');
        }
```

It would need to look like 

```
        // Select the item(s) and retrieve the id
       
        if ($isModule) {
            $query->select($db->quoteName('id'))
                  ->from($db->quoteName('#__modules'))
                ->where($db->quoteName('module') . ' = :extension');
        } else {
            $query->select($db->quoteName('extension_id') . ' as id')
                ->from($db->quoteName('#__extensions'))
                ->where($db->quoteName('element') . ' = :extension');
        }
```

**Comment by svanschu** [comment]
I'm working on it and will prepare a PR

**Comment by svanschu** [comment]
Fixed in PR  #42192

**Comment by svanschu** [comment]
PR  #42199
